exporters/otlp/otlptrace/otlptracehttp/client.go

```diff
@@ -154,28 +156,48 @@ func (d *client) UploadTraces(ctx context.Context, protoSpans []*tracepb.Resourc
 			return err
 		}
 
-		var rErr error
+		if resp != nil && resp.Body != nil {
+			defer func() {
+				if err := resp.Body.Close(); err != nil {
+					otel.Handle(err)
+				}
+			}()
+		}
+
```

Comment:
@MadVikingGod PTAL I tried to make the closing the response body and handling of errors more consistent based on your feedback.